lib/spack/spack/install_test.py

```diff
@@ -64,12 +64,16 @@ def valid_stage(d):
     return test_suites
 
 
-def get_test_suite(name):
+def get_named_test_suites(name):
+    """Return a list of the names of any test suites with that name."""
     assert name, "Cannot search for empty test name or 'None'"
     test_suites = get_all_test_suites()
-    names = [ts for ts in test_suites
-             if ts.name == name]
-    assert len(names) < 2, "alias shadows test suite hash"
+    return [ts for ts in test_suites if ts.name == name]
+
+
+def get_test_suite(name):
+    names = get_named_test_suites(name)
+    assert len(names) < 2, "Too many suites with that name.  May shadow hash."
```

Comment:
Similarly, this should probably raise some subclass of SpackError

Reply:
Definitely!  @tgamblin Should I change the  rest of the `asserts` in this file in this PR or a follow-up?

Reply:
Nvm.  I plan to change the rest of the asserts in the class in a separate PR.